### ISSUE TITLE
Triple-equals operator

### ISSUE THREAD

**Issue by atz** [projectblacklight/blacklight#1471]
There appears to be some confusion about the triple-equals/threequals operator.  Reference: [http://stackoverflow.com/questions/3422223/vs-in-ruby](http://stackoverflow.com/questions/3422223/vs-in-ruby
)

We use `===` many places:
```
app/helpers/blacklight/catalog_helper_behavior.rb:      if url_options === false || url_options[:suppress_link]
app/helpers/blacklight/facets_helper_behavior.rb:      localization_options = facet_config.date unless facet_config.date === true
app/presenters/blacklight/rendering/link_to_facet.rb:          return config.key if config.link_to_search === true
app/services/blacklight/field_retriever.rb:        if field_config.accessor === true
lib/blacklight/configuration/context.rb:        return config if config === true or config === false
lib/blacklight/configuration/fields.rb:        if @table[:luke_fields] === false
lib/blacklight/search_builder.rb:        if options.first === true
```

It works out for many values, because ruby has separately namespaced `TrueClass` and `FalseClass` (and those use `===` appropriately), like:
```ruby
> true===true
 => true 
> false===false
 => true 
> true===false
 => false 
```

But it also can cause problems.  For example, looking at `lib/blacklight/configuration/context.rb`, I can call `evaluate_if_unless_configuration(Object)` and break the logic and `Boolean` return type because:
```ruby
> Object===true
 => true 
> Object===false
 => true 
```

And if I pass it a `Proc`, the proc always executes and the `===` comparison is always true... until the Proc returns false but then is not threequal false anymore:
```ruby
> x=Proc.new { false }
 => #<Proc:0x007fba1186fee8@(irb):7> 
> x===true
 => false 
> x===false
 => false 
```

Basically, I see this as inadequate param checking and prone to confusion/failure.
